internal/app/machined/pkg/system/service_runner.go

```diff
@@ -224,7 +224,8 @@ func (svcrunner *ServiceRunner) Start() {
 		svcrunner.UpdateState(events.StateFinished, "Service finished successfully")
 	}
 
-	if err := svcrunner.service.PostFunc(svcrunner.config); err != nil {
+	// PostFunc passes in the state so that we can take actions that depend on the outcome of the run
+	if err := svcrunner.service.PostFunc(svcrunner.config, svcrunner.state); err != nil {
```

Comment:
`svcrunner.state` is protected with `svcrunner.mu` mutex (see `UpdateState` as an example)

so I think we might need `.GetState()` function with proper locking, and we should be all good